modules/terraform/workspace_test.go

```diff
@@ -130,3 +130,177 @@ func TestNameMatchesWorkspace(t *testing.T) {
 		assert.Equal(t, testCase.expected, actual, "Name: %q, Workspace: %q", testCase.name, testCase.workspace)
 	}
 }
+
+// Please note that this test depends on other functions that should be mocked to be a unit test.
+func TestWorkspaceDeleteE(t *testing.T) {
+	t.Parallel()
+
+	// state describes an expected status when a given testCase begins
+	type state struct {
+		workspaces []string
+		current    string
+	}
+
+	// testCase describes a named test case with a state, args and expcted results
+	type testCase struct {
+		name                 string
+		initialState         state
+		toDeleteWorkspace    string
+		expectedCurrent      string
+		expectedErrorMessage string
+	}
+
+	testCases := []testCase{
+		{
+			name: "delete another existing workspace and stay on current",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace:    "production",
+			expectedCurrent:      "staging",
+			expectedErrorMessage: "",
+		},
+		{
+			name: "delete current workspace and switch to a specified",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "production",
+			},
+			toDeleteWorkspace:    "production",
+			expectedCurrent:      "default",
+			expectedErrorMessage: "",
+		},
+		{
+			name: "delete a non existing workspace should trigger an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace:    "hellothere",
+			expectedCurrent:      "staging",
+			expectedErrorMessage: "The workspace \"hellothere\" does not exist.",
+		},
+		{
+			name: "delete the default workspace triggers an error",
+			initialState: state{
+				workspaces: []string{"staging", "production"},
+				current:    "staging",
+			},
+			toDeleteWorkspace:    "default",
+			expectedCurrent:      "staging",
+			expectedErrorMessage: "Deleting the workspace 'default' is not supported",
+		},
+	}
+
+	for _, tt := range testCases {
+		testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-workspace", tt.name)
```

Comment:
These should all be done within a `t.Run` so each test case can be executed separately.